### DISCUSSION TITLE
I cannot access my app.  The browser just hangs and then crashes.

### DISCUSSION THREAD

**Discussion by kpturner** [Budibase/budibase#11470] [Help]
I have made a considerable number of changes to our app today.   The final change I made was to a datasource containing a repeater where I change the limit to be 10000 rows and unticked pagination.   Now I can no longer access that screen. The browser hangs and crashes.  This means that I cannot publish it or get a useful usable export.  Please advise.

**Comment by kpturner** [comment]
In the end I had to revert to the latest export I had taken and then reapply all my changes again. Not ideal so looking for a better way?

This seems to be caused by a having a data provider on a table with 18000 records on it, with a filter that is dictated by a value in the URL.  So although there are 18000 records, when running "for real" the URL parameter results in the screen only loading the data it needs (usually about 3 or for rows).  However, in design mode, the URL does not contain anything.  Instead of displaying nothing it loads every single record - or tries to before blowing to smithereens.

**Comment by melohagan** [comment]
> change the limit to be 10000 rows and unticked pagination. Now I can no longer access that screen.

That's a bummer, but not unexpected. Pagination and/or limits are strongly recommended in the Design section for tables with many rows!

There is a hard limit of 5000 for SQL tables, but it could be a huge number of components being rendered.

In future you may be able to edit the JSON app export file, and manually adjust the limit.

**Reply by kpturner** [reply]
I did try that actually but it wasn't having any of it when I reimported it - so I thought that perhaps it was not the records/pagination after all.  It is a provider with a nested repeater and lots of components so I definitely won't try that again.  A useful feature would be a warning should you do this inadvertently.....or the designer handling the situation more elegantly (rather than leaving you dead in the water).

**Reply by melohagan** [reply]
Thanks for the feedback, I'll share it along.
